### PR TITLE
Knight tour ---> Closes : [#1849]

### PR DESCRIPTION
**Pull request by Prerna-07**
closes: #1849

### CODE REVIEW

**Comment by Prerna-07** [comment]
@nakul-19 @raksha009 @plazzy99 @tarun26091999 @rish-singhal @yasharth291 @Rohan27s @rudrakshi99 pls check this PR.

**Comment by Prerna-07** [comment]
@nakul-19 @raksha009 @plazzy99 @tarun26091999 @rish-singhal @yasharth291 @Rohan27s @rudrakshi99 pls check this PR.

**Comment by Prerna-07** [comment]
@nakul-19 @raksha009 @plazzy99 @tarun26091999 @rish-singhal @yasharth291 @Rohan27s @rudrakshi99 pls check this PR.

File: Algorithm/Backtracking/readme.md [hunk 1]
```diff
@@ -31,3 +31,4 @@ unsolvable.
 * Rat in a Maze ----> [C++](/Code/C++/rat_in_maze.cpp)
 * N Queen Problem ----> [C++](/Code/C++/n_queen_problem.cpp)
 * Sudoku Solver ----> [C++](/Code/C++/sudoku_solver.cpp) | [Java](/Code/Java/Sudoku_Solver.java)
```
Comment (rish-singhal) [inline Algorithm/Backtracking/readme.md:33]: Sort this list

File: Code/Java/Knights_Tour.java [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        int[][] chess = new int[n][n];
+        int row = scn.nextInt();                      // starting row
+        int col = scn.nextInt();                      // starting column
+        printKnightsTour(chess,row,col,1);            // calling function
+        
+    }
+
+    public static void printKnightsTour(int[][] chess, int row,int col,int count ) {
+       
+       if(row<0 || row>=chess.length || col<0 || col>=chess.length || chess[row][col]!=0)  // check for invalid moves i.e if box is already visited, or for non exixting points
```
Comment (rish-singhal) [inline Code/Java/Knights_Tour.java:27]: Give proper spaces around >,=,>>,<<,..... in whole file

File: Code/Java/Knights_Tour.java [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+19 10 5 16 25 
+6 15 18 11 4 
+1 20 9 24 17 
+14 7 22 3 12 
+21 2 13 8 23 
+        
+*/
+/*
+Time Complexity : O(8^(N^2))
```
Comment (rish-singhal) [inline Code/Java/Knights_Tour.java:416]: Describe what is N here

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Prerna-07** [comment]
@rish-singhal  done pls check.

**Comment by Prerna-07** [comment]
@rish-singhal  added extra line in this as well pls check.

File: Algorithm/Backtracking/readme.md [hunk 4]
```diff
@@ -27,6 +27,7 @@ unsolvable.
 
 ## Questions :
 
+* Knights Tour  ----> [Java](Code/Java/Knights_Tour.java)
 * Permutation of a string ----> [C++](/Code/C++/permutation_of_a_string.cpp)
 * Rat in a Maze ----> [C++](/Code/C++/rat_in_maze.cpp)
```
Comment (rish-singhal) [inline Algorithm/Backtracking/readme.md:31]: Sort this list

File: Code/Java/Knights_Tour.java [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+6 15 18 11 4 
+1 24 9 20 17 
+14 7 22 3 12 
+23 2 13 8 21 
+
+19 10 5 16 25 
+6 15 18 11 4 
+1 20 9 24 17 
+14 7 22 3 12 
+21 2 13 8 23
```
Comment (rish-singhal) [inline Code/Java/Knights_Tour.java:412]: Code seems good to go

File: Code/Java/Knights_Tour.java [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+        int[][] chess = new int[n][n];
+        int row = scn.nextInt();                      // starting row
+        int col = scn.nextInt();                      // starting column
+        printKnightsTour(chess,row,col,1);            // calling function
+        
+    }
+
+    public static void printKnightsTour(int[][] chess, int row,int col,int count ) {
+       
+       if(row < 0 || row>=chess.length || col < 0 || col >= chess.length || chess[row][col] != 0)  // check for invalid moves i.e if box is already visited, or for non exixting points
```
Comment (rish-singhal) [inline Code/Java/Knights_Tour.java:27]: Add space around >=,...

File: Code/Java/Knights_Tour.java [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+            displayBoard(chess);                    // print the result
+            chess[row][col] = 0;                      // undo the changes made in box so that next solution can be generated
+            return;
+        }
+        
+        
+        chess[row][col] = count;                     // if no base cases applies then put the count value in the currrent block
+       
+        
+       printKnightsTour(chess, row-2, col+1, count+1); // solution is searching clock wise all these functions are calling row and columns accordingly
```
Comment (rish-singhal) [inline Code/Java/Knights_Tour.java:45]: Here also add spaces around +,-,...

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Prerna-07** [comment]
@rish-singhal  done pls check

File: Code/Java/Knights_Tour.java (hunk 6 above)
Comment (rish-singhal) [reply Code/Java/Knights_Tour.java:27]: This is not resolved

**Comment by rish-singhal** [comment]
Just one more change

**Comment by Prerna-07** [comment]
@rish-singhal done pls check

**Pull request merged**